src/map_command_handler.hpp

```diff
@@ -191,9 +191,21 @@ class map_command_handler
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
 			symbols["command"] = get_cmd();
-			symbols["help_command"] = cmd_prefix_ + "help";
-			print("help", VGETTEXT("Unknown command '$command', try $help_command "
-				"for a list of available commands.", symbols));
+			std::string input = get_data(0);
+			if (cmd_prefix_ == "/"){
```

Comment:
Doing it like this still means that the class treats two prefix strings specially. Instead of hardcoding strings, please check whether `input` starts with `cmd_prefix_`. We're currently using C++17, so we don't have `std::string::starts_with()`, but we do have `boost::algorithm::starts_with()`. @CelticMinstrel do we have a utils function for that (I thought so, but can't find it)?

I think the logic should check whether get_command would succeed for the suggested change, so that `:help` would get a hint to remove the `cmd_prefix_`, but `:unknown` wouldn't.

Reply:
I'm not aware of one?

Even without Boost or C++17, "starts with" is just `str.find(x) == 0`, right?

Reply:
> Doing it like this still means that the class treats two prefix strings specially. 

Wait, isn't that what we want? We want the `"help_command"` symbol to be `cmd_prefix_ + "help"` when it's a chat command and just `"help"` when its a "normal" command, so we need to make a distinction (unless there is another way of checking whether you are in a chat command or in a "normal" command which I do not know of). Did you mean that I should be using something similar to `input[0] == cmd_prefix_` instead of `input[0] == ':'` in line 201?

> I think the logic should check whether get_command would succeed for the suggested change, so that :help would get a hint to remove the cmd_prefix_, but :unknown wouldn't.

I feel like that is reasonable.

Reply:
> Did you mean that I should be using something similar to input[0] == cmd_prefix_ instead of input[0] == ':' in line 201?

That's what I meant, although it might be multiple characters so it should be a starts-with check instead. Also, the translatable strings should put cmd_prefix_ in the symbols instead of including it in the string itself.

Reply:
e4506d5 should handle both requests.

Reply:
It still has a hardcoded `/`, and an assumption that cmd_prefix_ is exactly 1 character.